express-notes-api/index.js

```diff
@@ -0,0 +1,127 @@
+const express = require('express');
+
+const app = express();
+
+const data = require('./data.json');
+
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const arrayObjects = [];
+  const notes = data.notes;
+  for (const key in notes) {
+    arrayObjects.push(notes[key]);
+  }
+  res.send(arrayObjects);
+});
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  const number = Number(req.params.id);
+  if (number <= 0 || !Number.isInteger(number) || number === 'NaN') {
+    res.status(400).send({
+      error: 'id must be a postiive integer'
+    });
+  }
+  if (notes[number] === undefined) {
+    res.status(404).json({
+      error: 'no note with the specified id'
+    });
+  }
+  if (notes[number] !== undefined) {
+    res.status(200).json(notes[number]);
+  }
```

Comment:
`res.send()` does not exit the middleware function on its own, so make sure you find some way to prevent multiple responses from being sent.